### PR TITLE
fix: add missing cross-page revalidation paths

### PR DESCRIPTION
**Pull request by tonyrosario**
Adds revalidatePath("/dashboard") to 17 server actions across profile, groups, events, learning, and tracker so dashboard reflects mutations made on other pages without requiring a manual refresh.

Adds revalidatePath("/members") and revalidatePath("/events") to updateProfile, which previously had zero revalidations. ~~Removes revalidatePath("/profile") — it caused a hydration mismatch on the profile page (Radix UI PopoverTrigger injects aria-expanded into the client DOM; an RSC push during interactive state triggers reconciliation failure).~~ agent captured this removal of my own local test code, this was NOT in the body of work.

Tracker received 6 revalidation fixes (createApplication, updateApplication, updateStatusDate, deleteApplication, addInterview, deleteInterview) vs. 3 planned — all correct fixes in the same category.

Adds 5 new test files asserting correct revalidation paths and auth guards for all modified actions.

* * *
**Known limitations in this PR:**

1. /events timezone - **partial fix**. revalidatePath("/events") busts the server cache but not the client-side Router Cache. Soft navigation serves stale data; hard refresh (Cmd+R) required to see the new timezone. Root cause: Next.js client Router Cache TTL.

2. Dashboard timezone display - **by design, not fixed**. UpcomingEventsCard uses e.timezone || viewerTimezone — event-timezone-first. Since all events have a non-null timezone (since migration 057), viewer timezone is never reached. Changing your profile timezone has no effect on dashboard event times. **Product decision needed:** should it align with /events list (viewer-first)?

  **Follow-up items for next phase:**

3. Optimistic state diverges from server bugs still open:
  - Client components that initialize useState from props don't re-sync when the server re-renders — fresh data arrives but local state is ignored.
  - WeeklyScheduleAdmin and group notes in GroupHubClient copy props into useState and ignore fresh server data after router.refresh(). Not touched by this PR.

4. React correctness bugs still open:
  - router.refresh() inside setConversations updater
  - hydration mismatch in group tabs
  - nested <button> in GroupHubClient.

### CODE REVIEW

File: app/(app)/groups/actions.ts [hunk 1]
```diff
@@ -43,6 +43,7 @@ export async function createGroupAction(
       input.isDiscoverable
     );
     revalidatePath("/groups");
+    revalidatePath("/dashboard");
```
Comment (kaizengrowth) [inline app/(app)/groups/actions.ts:46]: Nice! LGTM

File: app/(app)/events/actions.test.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+jest.mock("@/lib/supabase/server", () => ({ createClient: jest.fn() }));
+
+import { revalidatePath } from "next/cache";
+import { createClient } from "@/lib/supabase/server";
+import { deleteEvent } from "./actions";
+
+const mockRevalidatePath = revalidatePath as jest.MockedFunction<typeof revalidatePath>;
+const mockCreateClient = createClient as jest.MockedFunction<typeof createClient>;
+
+function makeDeleteChain() {
```
Comment (kaizengrowth) [inline app/(app)/events/actions.test.ts:16]: NIT:  Similar `makeChain` helpers exist in many test files, so this might be good as a helper function in a shared test utility file.

File: app/(app)/events/actions.ts [hunk 3]
```diff
@@ -220,5 +220,6 @@ export async function deleteEvent(eventId: string) {
 
   if (error) throw new Error(error.message);
   revalidatePath("/events");
+  revalidatePath("/dashboard");
```
Comment (kaizengrowth) [inline app/(app)/events/actions.ts:223]: LGTM! Makes sense to have this after the /events revalidation and before the redirect.

File: app/(app)/groups/actions.test.ts [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  createGroupAction,
+  joinGroupAction,
+  leaveGroupAction,
+  updateGroupSettingsAction,
+} from "./actions";
+
+const mockRevalidatePath = revalidatePath as jest.MockedFunction<typeof revalidatePath>;
+const mockCreateClient = createClient as jest.MockedFunction<typeof createClient>;
+
+function makeClient(userId: string | null) {
```
Comment (kaizengrowth) [inline app/(app)/groups/actions.test.ts:27]: NIT: Can this be a reusable helper function?

File: app/(app)/groups/actions.test.ts [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+  beforeEach(() => { jest.clearAllMocks(); });
+
+  it("should return an error when user is not authenticated", async () => {
+    // Arrange
+    mockCreateClient.mockResolvedValue(makeClient(null) as any);
+
+    // Act
+    const result = await updateGroupSettingsAction("group-1", { is_discoverable: true });
+
+    // Assert
```
Comment (kaizengrowth) [inline app/(app)/groups/actions.test.ts]: Missing test case for the happy path with `newSlug` return?

File: app/(app)/learning/learning-actions.ts [hunk 6]
```diff
@@ -144,5 +150,6 @@ export async function deleteResource(id: string) {
   const { error } = await supabase.from("learning_resources").delete().eq("id", id);
   if (error) return { error: error.message };
   revalidatePath("/learning");
+  revalidatePath("/dashboard");
```
Comment (kaizengrowth) [inline app/(app)/learning/learning-actions.ts:153]: LGTM!

File: app/(app)/profile/actions.ts [hunk 7]
```diff
@@ -53,6 +53,9 @@ export async function updateProfile(
     return { error: error.message };
   }
 
+  revalidatePath("/members");
+  revalidatePath("/events");
+  revalidatePath("/dashboard");
   return {};
```
Comment (kaizengrowth) [inline app/(app)/profile/actions.ts:59]: The PR description says there was a `revalidatePath("/profile")` that was removed. Was it removed in a previous PR?

File: app/(app)/tracker/actions.test.ts [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+    const result = await updateStatusDate("app-1", "applied", "2026-03-15");
+
+    // Assert
+    expect(result).toEqual({});
+    expect(mockRevalidatePath).toHaveBeenCalledWith("/tracker");
+    expect(mockRevalidatePath).toHaveBeenCalledWith("/dashboard");
+  });
+});
+
+describe("deleteApplication — cache revalidation", () => {
```
Comment (kaizengrowth) [inline app/(app)/tracker/actions.test.ts]: Is there some missing test coverage for `createApplication`, `addInterview`, and `deleteInterview`?

**Review by kaizengrowth — DISMISSED** [review]
This PR looks like it correctly addresses the cross-page cache invalidation issue. Some minor suggestions for more complete test coverage. 

Really appreciate the thoughtful PR description as well, with consideration for edge cases.

File: app/(app)/profile/actions.ts (hunk 7 above)
Comment (tonyrosario) [reply app/(app)/profile/actions.ts:59]: Good catch - I missed that.

I was testing an automatic refresh (one tab on the dashboard, one tab on profile), an incorrect assumption on my part, and I had to remove it so it got flagged as a piece of work.  I'll update the description.

File: app/(app)/tracker/actions.test.ts (hunk 8 above)
Comment (tonyrosario) [reply app/(app)/tracker/actions.test.ts]: I'll double-check for all of the missing test coverage.

Comment (tonyrosario) [reply app/(app)/tracker/actions.test.ts]: New tests have been applied.

File: app/(app)/groups/actions.test.ts (hunk 4 above)
Comment (tonyrosario) [reply app/(app)/groups/actions.test.ts:27]: Agreed but I have the helpers slated for after the testing standards get put in.  Then we can build out the shared mock infrastructure.

Sound fair?

File: app/(app)/events/actions.test.ts (hunk 2 above)
Comment (tonyrosario) [reply app/(app)/events/actions.test.ts:16]: Same as previous reply - we sure can but I ask that we add that in a separate PR.

File: app/(app)/profile/actions.ts (hunk 7 above)
Comment (tonyrosario) [reply app/(app)/profile/actions.ts:59]: updated PR description and commit message!

File: app/(app)/groups/actions.test.ts (hunk 5 above)
Comment (tonyrosario) [reply app/(app)/groups/actions.test.ts]: Added in adaaca98ba27d32d1deb9fb0946f054d5282a8f0

**Review by kaizengrowth — APPROVED** [review]
